src/game/game.c

```diff
@@ -25,14 +25,45 @@ bool StartGame(int32_t level_num, GAMEFLOW_LEVEL_TYPE level_type)
 {
     g_CurrentLevel = level_num;
     g_GameInfo.current_level_type = level_type;
-    if (level_type == GFL_SAVED) {
-        // reset start info to the defaults so that we do not do
+    if (level_type == GFL_SAVED || level_type == GFL_SELECT) {
+        // reset current info to the defaults so that we do not do
         // GlobalItemReplace in the inventory initialization routines too early
-        Savegame_ResetStartInfo(level_num);
+        Savegame_InitCurrentInfo();
     } else {
         InitialiseLevelFlags();
     }
 
+    if (level_type == GFL_SELECT) {
```

Comment:
Wasn't sure how best to do this. Since `GFL_SAVED` needs to happen after `InitialiseLevel` while `GFL_SELECT` and `GFL_RESTART` need to go before. Should I do a switch just for select and restart?

Reply:
No, I had in mind a separate `case` for each `level_type` where each case has specifies its own flow of things in its entirety.

Reply:
Done.